src/slurm_plugin/clustermgtd.py

```diff
@@ -639,34 +580,26 @@ def _handle_health_check(
         # Pick corresponding health check function
         is_instance_unhealthy = health_check_functions.get(health_check_type).get("function")
         default_kwargs = health_check_functions.get(health_check_type).get("default_kwargs")
-        for instance in unhealthy_instance_status:
-            if is_instance_unhealthy(instance_health_state=instance, **default_kwargs):
-                unhealthy_node = ip_to_slurm_node_map.get(id_to_instance_map.get(instance.id).private_ip)
-                if unhealthy_node:
-                    log.warning(
-                        "Node %s(%s) is associated with instance %s that is failing %s. EC2 health state: %s",
-                        unhealthy_node.name,
-                        unhealthy_node.nodeaddr,
-                        instance.id,
-                        health_check_type,
-                        instance,
-                    )
-                    if unhealthy_node.name in self._static_nodes_in_replacement:
-                        log.warning(
-                            "Detected failed health check for static node in replacement. "
-                            "No longer considering node %s(%s) as in replacement process. "
-                            "Will attempt to replace node again immediately.",
-                            unhealthy_node.name,
-                            unhealthy_node.nodeaddr,
-                        )
-                        self._static_nodes_in_replacement.remove(unhealthy_node.name)
-                    nodes_failing_health_check.append(unhealthy_node.name)
-        if nodes_failing_health_check:
-            # Place unhealthy node into drain, this operation is idempotent
-            log.warning(
-                "Setting nodes failing health check type %s to DRAIN: %s", health_check_type, nodes_failing_health_check
-            )
-            set_nodes_drain(nodes_failing_health_check, reason=f"Node failing {health_check_type}")
+        for instance_status in unhealthy_instances_status:
+            unhealthy_node = instance_id_to_active_node_map.get(instance_status.id)
+            if unhealthy_node and is_instance_unhealthy(instance_status, **default_kwargs):
```

Comment:
Similar to previous command, I don't think you can pass `instance_status` into a non-static method like this.

Reply:
Replied in the comments above